.github/workflows/check-static-files.yml

```diff
@@ -0,0 +1,51 @@
+name: Check changes to static files
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  changed_files:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: List changed frontend source files
+        uses: tj-actions/changed-files@v44
+        id: changed-frontend-src-files
+        with:
+          files: |
+             frontend/src/**/*.tsx
+             frontend/src/**/*.css
+
+      - name: List changed frontend static files
+        id: changed-frontend-static-files
+        if: steps.changed-frontend-src-files.outputs.files_changed == 'true'
+        uses: tj-actions/changed-files@v44
```

Comment:
Per GitHub security recommendations, all (3rd-party) actions should be pinned to commit hash because these version tags are mutable.

Consider adding codeql to this repo.